### PR TITLE
Rename ChannelPersistence to ConnectionPersistence.

### PR DESCRIPTION
**Pull request by A-Imal**
https://hivemq.kanbanize.com/ctrl_board/4/cards/5375/details/

### CODE REVIEW

File: src/main/java/com/hivemq/mqtt/handler/connect/MessageBarrier.java [hunk 1]
```diff
@@ -54,8 +50,8 @@ public class MessageBarrier extends ChannelDuplexHandler {
     private final @NotNull MqttServerDisconnector serverDisconnector;
     private final @NotNull Queue<Message> messageQueue = new LinkedList<>();
 
-    private boolean connectReceived = false;
-    private boolean connackSent = false;
+    private boolean connectReceived;
+    private boolean connackSent;
```
Comment (Remit) [inline src/main/java/com/hivemq/mqtt/handler/connect/MessageBarrier.java:54]: will the above two be set to false by default? IMHO, even if they are, it makes sense to make it explicit in the ctor

File: src/main/java/com/hivemq/mqtt/services/PublishPollServiceImpl.java [hunk 2]
```diff
@@ -122,29 +117,25 @@ public void pollMessages(final @NotNull String client, final @NotNull Channel ch
...
-     */
     @Override
     public void pollNewMessages(final @NotNull String client) {
-        final Channel channel = channelPersistence.get(client);
-        if (channel == null) {
-            return; // client is disconnected
+        final ClientConnection clientConnection = connectionPersistence.get(client);
+        if (clientConnection == null
+                || clientConnection.getClientState() == ClientState.DISCONNECTING
+                || clientConnection.getClientState().disconnected()) {
```
Comment (Remit) [inline src/main/java/com/hivemq/mqtt/services/PublishPollServiceImpl.java:133]: just to make sure - this is a change from some other PR, not from this refactoring?

File: src/main/java/com/hivemq/security/auth/ClientToken.java [hunk 3]
```diff
@@ -181,15 +169,15 @@ public boolean equals(final Object o) {
         if (clientId != null ? !clientId.equals(that.clientId) : that.clientId != null) return false;
         if (listener != null ? !listener.equals(that.listener) : that.listener != null) return false;
         if (username != null ? !username.equals(that.username) : that.username != null) return false;
-        if ((password.isPresent() && !that.password.isPresent()) || (!password.isPresent() && that.password.isPresent())) {
+        if (password.isPresent() ? !that.password.isPresent() : that.password.isPresent()) {
```
Comment (Remit) [inline src/main/java/com/hivemq/security/auth/ClientToken.java:172]: so, basically that's a XOR? https://www.baeldung.com/java-xor-operator
I'd suggest to use ^ then, it clearly shows the intention that both passwords are different.

File: src/main/java/com/hivemq/security/auth/ClientToken.java [hunk 4]
```diff
@@ -116,15 +112,7 @@ public String getClientId() {
...
-        return password.transform(new Function<byte[], String>() {
-            @Override
-            public String apply(final byte[] input) {
-                if (input == null) {
-                    return null;
-                }
-                return new String(input, StandardCharsets.UTF_8);
-            }
-        });
+        return password.map((input) -> new String(input, StandardCharsets.UTF_8));
```
[the remark this answers is not in the document]
Comment (A-Imal) [reply src/main/java/com/hivemq/security/auth/ClientToken.java:115]: Yes, and it also doesn't matter. My next PR removes this class entirely.

File: src/main/java/com/hivemq/security/auth/ClientToken.java [hunk 5]
```diff
@@ -181,15 +169,15 @@ public boolean equals(final Object o) {
...
-        if ((password.isPresent() && !that.password.isPresent()) || (!password.isPresent() && that.password.isPresent())) {
+        if (password.isPresent() ? !that.password.isPresent() : that.password.isPresent()) {
             return false;
         }
         if ((password.isPresent() && that.password.isPresent()) && !Arrays.equals(password.get(), that.password.get())) {
             return false;
         }
         if (certificate != null ? !certificate.equals(that.certificate) : that.certificate != null) return false;
         if (inetAddress != null ? !inetAddress.equals(that.inetAddress) : that.inetAddress != null) return false;
-        if ((listener.isPresent() && !that.listener.isPresent()) || (!listener.isPresent() && that.listener.isPresent()))
```
Comment (Remit) [inline src/main/java/com/hivemq/security/auth/ClientToken.java:192]: XOR? ^

File: src/main/java/com/hivemq/mqtt/handler/connect/MessageBarrier.java (hunk 1 above)
Comment (A-Imal) [reply src/main/java/com/hivemq/mqtt/handler/connect/MessageBarrier.java:54]: Yes, Java will initialize boolean per default with false.

**Review by Remit — APPROVED** [review]

**Pull request merged**